Classes/git/PBGitIndex.m

```diff
@@ -521,8 +521,24 @@ - (BOOL)applyPatch:(NSString *)hunk stage:(BOOL)stage reverse:(BOOL)reverse;
 		return NO;
 	}
 
-	// TODO: Try to be smarter about what to refresh
-	[self refresh];
+	// Get the status for the updated file and update the index
+	BOOL success = NO;
+	GTFileStatusFlags status = [[[self repository] gtRepo] statusForFile:[file path] success:&success error:NULL];
+
+	if (success) {
+		// Search for file in [self files]
+		// file is probably a member of [self files], but let's not assume that to be safe
+		for (PBChangedFile *nextFile in [self files]) {
+			if (nextFile == file || [[nextFile path] isEqualToString:[file path]]) {
```

Comment:
Not fond of this loop. I think something like this would be clearer :
```
idx = [self.files indexOfObjectPassingTest:/* predicate */];
if (idx != NSNotFound) {
  /* update flags for self.files[idx] */
}
```

Reply:
Done